src/middlewares/asyncWrapper.js

```diff
@@ -5,12 +5,15 @@
   * @function
   *
   * @param {Function} wrappedFunction - the main controller
+  * @param {Boolean} middleware - A flag to know if it is a middleware
   *
   * @return {Function} - callback that execute the controller
   */
-export default wrappedFunction => async (request, response, next) => {
+export default (wrappedFunction, middleware = false) => async (request, response, next) => {
```

Comment:
Niceee

Reply:
But this is going to be a middleware anyways (it takes in the request, response and next). I wonder why we need this?